### PR TITLE
doc: Add a test for the instance manager NPE issue

### PR DESCRIPTION
**Pull request by shuo-wu**
Longhorn/longhorn#3466

### CODE REVIEW

File: docs/content/manual/release-specific/v1.2.4/test-instance-manager-npe.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+### Test step
+1. Create and attach a 1-replica volume.
+2. Create 2 snapshots with large amount of data so that rebuilding each snapshot would take some time.
+3. Disable the scheduling for the nodes so that there is one node could accept new replicas of the volume.
+4. Update the replica count to 2 for the volume and wait for the rebuilding start.
+5. While syncing the 1st snapshot file, create a directory with the name of **another snapshot meta file**. 
+   Later the rebuilding replica will fail to create this meta file then error out.
+6. Verify there is no NPE issue in the instance manager pod when the failure mentioned above is triggered.
+7. Verify the rebuilding will be restarted and succeed.
+
```
Comment (khushboo-rancher) [inline docs/content/manual/release-specific/v1.2.4/test-instance-manager-npe.md:14]: 8. Verify the data of the volume.

**Review by khushboo-rancher — CHANGES_REQUESTED** [review]

File: docs/content/manual/release-specific/v1.2.4/test-instance-manager-npe.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+---
+
+### Test step
+1. Create and attach a 1-replica volume.
+2. Create 2 snapshots with large amount of data so that rebuilding each snapshot would take some time.
+3. Disable the scheduling for the nodes so that there is one node could accept new replicas of the volume.
+4. Update the replica count to 2 for the volume and wait for the rebuilding start.
+5. While syncing the 1st snapshot file, create a directory with the name of **another snapshot meta file**. 
+   Later the rebuilding replica will fail to create this meta file then error out.
+6. Verify there is no NPE issue in the instance manager pod when the failure mentioned above is triggered.
```
[the remark this answers is not in the document]
Comment (shuo-wu) [reply docs/content/manual/release-specific/v1.2.4/test-instance-manager-npe.md]: Sure. I have added more detail for this part. PTAL.

**Comment by shuo-wu** [comment]
I am thinking if this test doc should be put into `v1.3.0` instead.

**Review by khushboo-rancher — DISMISSED** [review]
LGTM, Yes `v1.3.0` would make sense.

File: docs/content/manual/release-specific/v1.2.4/test-instance-manager-npe.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+---
+title: Test instance manager NPE 
+---
+
+### Test step
+1. Create and attach a 1-replica volume.
+2. Create 2 snapshots with large amount of data so that rebuilding each snapshot would take some time.
+3. Disable the scheduling for the nodes so that there is one node could accept new replicas of the volume.
+4. Update the replica count to 2 for the volume and wait for the rebuilding start.
+5. While syncing the 1st snapshot file, create a directory with the name of **another snapshot meta file**.
```
Comment (innobead) [inline docs/content/manual/release-specific/v1.2.4/test-instance-manager-npe.md:10]: Ok, so this is to simulate the receiver failing to create the file to cause the sender to close the sending server. 

Before step 6, should we delete the intensively created directory of the second snapshot meta file to make sure the following rebuilding is successful?

**Comment by innobead** [comment]
> LGTM, Yes `v1.3.0` would make sense.

@shuo-wu would you like to update this?

File: docs/content/manual/release-specific/v1.2.4/test-instance-manager-npe.md (hunk 3 above)
Comment (shuo-wu) [reply docs/content/manual/release-specific/v1.2.4/test-instance-manager-npe.md:10]: No. We are intended to fail this rebuilding. And this messed rebuilding replica would be cleaned up after the failure hence we don't need to do anything here.

**Review by khushboo-rancher — APPROVED** [review]

**Pull request merged**